src/prefect/_internal/concurrency/api.py

```diff
@@ -231,6 +233,24 @@ def wait_for_call_in_loop_thread(
         done_callbacks: Optional[Iterable[Call]] = None,
         contexts: Optional[Iterable[ContextManager]] = None,
     ) -> Awaitable[T]:
+        # Check for a running event loop to prevent blocking
+        if get_running_loop():
+            if not greenback.has_portal():
+                raise RuntimeError(
+                    "Detected unsafe call to `from_sync` from thread with event loop. "
+                    "Call `await greenback.ensure_portal()` to allow functionality."
+                )
+
+            # Use greenback to avoid blocking the event loop while waiting
+            return greenback.await_(
+                from_async.wait_for_call_in_loop_thread(
+                    __call,
+                    timeout=timeout,
+                    done_callbacks=done_callbacks,
+                    contexts=contexts,
+                )
+            )
```

Comment:
Wow isn't this magical!

Here, we avoid blocking the event loop in a `from_sync` call from a thread with an event loop by using `greenback` to schedule the call as a `from_async` waiter instead!

This error message is not intended to be user-facing.